rdwatch/core/views/animation.py

```diff
@@ -0,0 +1,218 @@
+    GenerateAnimationSchema,
+    create_animation,
+    create_modelrun_animation_export,
+    create_site_animation_export,
+)
+
+logger = logging.getLogger(__name__)
+
+router = Router()
+
+
+def generate_download_data(
+    exports: list[AnimationSiteExport] | list[AnimationModelRunExport],
+):
+    output = []
+    for export in exports:
+        if export.export_file:  # File exists it can be downloaded
+            output.append(
+                {
+                    'created': export.created,
+                    'completed': True,
+                    'arguments': export.arguments,
+                    'taskId': export.celery_id,
+                }
+            )
+        else:
+            task_id = export.celery_id
+            task = AsyncResult(task_id)
+            celery_data = {}
+            celery_data['state'] = task.state
+            celery_data['status'] = task.status
+            celery_data['info'] = (
+                str(task.info) if isinstance(task.info, RuntimeError) else task.info
+            )
+            output.append(
+                {
+                    'created': export.created,
+                    'completed': False,
+                    'arguments': export.arguments,
+                    'taskId': export.celery_id,
+                    'state': celery_data,
+                }
+            )
+    return output
+
+
+@router.post('/site/{id}/')
+def generate_site_animation(
+    request: HttpRequest,
+    id: UUID4,
+    params: GenerateAnimationSchema,  # noqa: B008
+):
+    # Fetch the SiteEvaluation instance
+    task_id = create_site_animation_export.delay(
+        site_evaluation_id=id, settings=params.dict(), userId=request.user.pk
+    )
+    return task_id.id
+
+
+@router.get('/site/{id}/downloads/')
+def get_site_downloads(
+    request: HttpRequest,
+    id: UUID4,
+):
+    exports = AnimationSiteExport.objects.filter(
+        site_evaluation=id, user=request.user
+    ).order_by('-created')
+    return generate_download_data(exports)
+
+
+@router.get('/modelrun/{id}/downloads/')
+def get_modelrun_downloads(
+    request: HttpRequest,
+    id: UUID4,
+):
+    exports = AnimationModelRunExport.objects.filter(
+        configuration=id, user=request.user
+    ).order_by('-created')
+    return generate_download_data(exports)
+
+
+class DeleteErrorSchema(BaseModel):
+    error: str
+
+
+class DeleteSuccessSchema(BaseModel):
+    success: str
+
+
+@router.delete(
+    '/site/{id}/',
+    response={
+        403: DeleteErrorSchema,
+        200: DeleteSuccessSchema,
+    },
+)
+def delete_site_download(
+    request: HttpRequest,
+    id: UUID4,
+):
+    # Fetch the export object
+    export = get_object_or_404(AnimationSiteExport, celery_id=id, user=request.user)
+    logger.warning(export)
+    # Check for ownership or permissions
+    if export.user != request.user:
+        return 403, {'error': 'You do not have permission to delete this export.'}
```

Comment:
Since you're including `user` in the `get_object_or_404` query, the `export` will always belong to this user, or a 404 will be returned. 

I'm not sure if that was intentional or not, but I think that behavior is fine. I actually prefer this pattern over returning a 403, since returning a 403 to an unauthorized user is actually leaking the existence of the object to them. not that it matters much in this specific case, but in general I usually go with this strategy.

fwiw, github does the same thing; notice it will return a 404 if you try to access a private repo while logged out.

```suggestion
    # Fetch the export object
    export = get_object_or_404(AnimationSiteExport, celery_id=id, user=request.user)
```